interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+    builder.addResponseParameters(ResponseParameters.newBuilder().setSize(streamSize));
+    StreamingOutputCallRequest request = builder.build();
+
+    TestStreamObserver observer = new TestStreamObserver();
+    stub.streamingOutputCall(request, observer);
+    try {
+      observer.waitFor();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    // Don't actually need bwithUsed right now but keeping it around because it might be interesting
+    long bwithUsed = (streamSize / (observer.getElapsedTime() / TimeUnit.SECONDS.toNanos(1)));
+    long expectedWindow = (latency * (bandwidth / TimeUnit.SECONDS.toMillis(1))) * 2;
+    int lastWindow = logHandler.getLastWindow();
+
+    // deal with cases that either don't cause a window update or hit max window
+    expectedWindow = Math.min(MAX_WINDOW, (Math.max(expectedWindow, REGULAR_WINDOW)));
+
+    // I know this range looks large, but this allows for only one extra/missed window update
+    // (one extra update causes a 2x difference and one missed update causes a .5x difference)
+    if ((lastWindow < 2.1 * expectedWindow) && (expectedWindow < 2.1 * lastWindow)) {
+      System.out.println("PASS");
+      testsPassed++;
+    } else {
+      System.out.println("FAIL");
+      testsFailed++;
+    }
+  }
+
+  /**
+   * Resets client/server and their flow control windows
+   *
+   * @param serverFlowControlWindow
+   * @param clientFlowControlWindow
+   */
+  private void resetConnection(int serverFlowControlWindow, int clientFlowControlWindow) {
+    if (channel != null) {
+      if (!channel.isShutdown()) {
+        channel.shutdown();
+      }
+    }
+    startServer(serverFlowControlWindow);
+
+    int channelPort = port;
+    if (useProxy) {
+      channelPort = proxyPort;
+    }
+    channel = NettyChannelBuilder.forAddress(new InetSocketAddress(loopback, channelPort))
+        .flowControlWindow(clientFlowControlWindow).negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private void startServer(int serverFlowControlWindow) {
+    ServerBuilder<?> builder = NettyServerBuilder.forAddress(new InetSocketAddress(loopback, port))
+        .flowControlWindow(serverFlowControlWindow);
+    List<ServerInterceptor> allInterceptors = ImmutableList.<ServerInterceptor>builder()
+        .build();
+    builder.addService(ServerInterceptors.intercept(
+        TestServiceGrpc.bindService(new TestServiceImpl(Executors.newScheduledThreadPool(2))),
+        allInterceptors));
+    try {
+      server = builder.build().start();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Restart proxy with new bandwidth/latency settings.
+   *
+   * @param targetBPS
+   * @param targetLatency
+   * @param latencyUnits
+   */
+  private void resetProxy(final int targetBPS, final int targetLatency,
+      final TimeUnit latencyUnits) {
+    executor.submit(new Runnable() {
```

Comment:
Returned the future, which required a change in the proxy. Before, start did not terminate until it spun up workers, which happens after the proxy connects to server/client. Now, there is an additional worker that waits for and accepts connections